leakcanary-analyzer/src/main/java/leakcanary/internal/ShortestPathFinder.kt

```diff
@@ -145,19 +174,82 @@ internal class ShortestPathFinder {
       val weakReference = referentMap[node.instance]
       if (weakReference != null) {
         val exclusionPriority = if (lowestPriority == ALWAYS_REACHABLE) null else lowestPriority
-        results.add(Result(node, exclusionPriority, weakReference))
-        // Found all refs, stop searching.
-        if (results.size == leakingWeakRefs.size) {
-          break
+        resultsFound.add(Result(node, exclusionPriority, weakReference, null))
+        // Found all refs, stop searching (unless computing retained size which stops on weak reachables)
+        if (resultsFound.size == leakingWeakRefs.size) {
+          if (computeRetainedHeapSize && lowestPriority < WEAKLY_REACHABLE) {
```

Comment:
TODO This isn't correct, we're actually going through all instances because we never check this condition again (no more leaking instances found)